search/terms_aggregation.go

```diff
@@ -0,0 +1,201 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// TermsAggregation dynamically creates a bucket for each unique term of a field.
+// An empty TermsAggregation will have two issues with execution:
+//
+//   - the target Field must be non-null and non-empty.
+//   - a Size of 0 will return no buckets
+//
+// For more details see https://opensearch.org/docs/latest/opensearch/bucket-agg/
+type TermsAggregation struct {
+	// Field to be bucketed
+	Field string
+
+	// Size of the number of buckets to be returned. Negative sizes will be omitted
+	Size int
+
+	// Order list of [Order]s to sort the aggregation buckets. Default order is _count: desc
+	Order []Order
+
+	// Aggregations sub aggregations for each bucket. Mapped by string label to sub aggregation
+	Aggregations map[string]Aggregation
+}
+
+// NewTermsAggregation instantiates a TermsAggregation targeting the provided field
+// and sets the Size to -1 to be omitted for the default value.
+func NewTermsAggregation(field string) *TermsAggregation {
+	return &TermsAggregation{
+		Field:        field,
+		Size:         -1,
+		Aggregations: make(map[string]Aggregation),
+	}
+}
+
+// WithSize for the number of buckets to be returned
+func (t *TermsAggregation) WithSize(size int) *TermsAggregation {
+	t.Size = size
+	return t
+}
+
+// AddOrder of the returned buckets
+func (t *TermsAggregation) AddOrder(orders ...Order) *TermsAggregation {
+	t.Order = append(t.Order, orders...)
+	return t
+}
+
+// AddSubAggregation to the TermsAggregation with the provided name
+func (t *TermsAggregation) AddSubAggregation(name string, agg Aggregation) BucketAggregation {
+	if t.Aggregations == nil {
```

Comment:
this check will not require. as we initialise in NewTermsAggregation

Reply:
A consumer is not required to use `NewTermsAggregation`. They could instantiate with a literal, which wouldn't instantiate the map.
```
agg := search.TermsAggregation{}
```